pkgs/development/libraries/qt-5/5.11/qtbase-additional.patch

```diff
@@ -0,0 +1,32 @@
+diff --git a/src/corelib/kernel/qcoreapplication.cpp b/src/corelib/kernel/qcoreapplication.cpp
+index 5ae3fd62e5..ec3fccfc76 100644
+--- a/src/corelib/kernel/qcoreapplication.cpp
++++ b/src/corelib/kernel/qcoreapplication.cpp
+@@ -2533,6 +2533,27 @@ QStringList QCoreApplication::libraryPaths()
+         QStringList *app_libpaths = new QStringList;
+         coreappdata()->app_libpaths.reset(app_libpaths);
+ 
++		{
++			// Start at the binary; this allows us to *always* start by stripping the last part.
++			QStringList components = applicationFilePath().split(QDir::separator());
++
++			// We don't care about /nix/store/nix-support, only /nix/store/*/nix-support
```

Comment:
I am uncomfortable with searching upward from the executable in this way. Going up one level should be fine (from `/nix/store/.../bin/foo` to `/nix/store/.../nix-support/`). The part about `components.length() > 3` smells fishy to me; what happens if Nix is built with a different store path, such as a user-mode installation?